### PR TITLE
Fix unknown threshold template

### PR DESCRIPTION
**Pull request by masha1997**

### CODE REVIEW

File: provider/resource_service_test.go [hunk 1]
```diff
@@ -251,41 +329,6 @@ func TestAccResourceServiceDependsOnLifecycle(t *testing.T) {
 		},
 	})
 }
-func testCheckServiceDependsOnMatch(child_name string, expected_overloaded_urgency ...int) resource.TestCheckFunc {
```
Comment (masha1997) [inline provider/resource_service_test.go:254]: on the line 203, i didn't delete it

File: Makefile [hunk 2]
```diff
@@ -39,7 +39,7 @@ test: fmt
 	go test -v -cover -parallel=4 $(TEST_ARGS) github.com/tivo/terraform-provider-splunk-itsi/... -tags test_setup
 
 testacc: fmt
-	TF_ACC=1 go test -v -cover $(TEST_ARGS) -timeout 10m ./...
+	TF_ACC=1 go test -v -cover $(TEST_ARGS) -timeout 30m ./...
```
Comment (masha1997) [inline Makefile:42]: it was already near 10 m, extended with adding a new test

**Comment by masha1997** [comment]
Plan after scrapper is clean:
[1mPlan:[0m 4503 to import, 0 to add, 0 to change, 0 to destroy.

File: provider/resource_service_test.go [hunk 3]
```diff
@@ -200,6 +200,84 @@ func TestAccResourceServiceTagsLifecycle(t *testing.T) {
 	})
 }
 
+func testCheckServiceDependsOnMatch(child_name string, expected_overloaded_urgency ...int) resource.TestCheckFunc {
```
Comment (masha1997) [inline provider/resource_service_test.go:203]: it's not the new test, this one was moved, bad diff

File: provider/resource_service_test.go [hunk 4]
```diff
@@ -200,6 +200,84 @@ func TestAccResourceServiceTagsLifecycle(t *testing.T) {
...
+				// }
+			}
+			return nil
+		}
+		return fmt.Errorf("shkpi_id mismatch: Missing shkpi_id %s\n", leafKPIID)
+	}
+
+}
+
+func TestAccResourceServiceKpisHandleUnknownTemplateId(t *testing.T) {
```
Comment (masha1997) [inline provider/resource_service_test.go:239]: this one is new to verify problem case

File: provider/resource_service.go [hunk 5]
```diff
@@ -402,17 +394,17 @@ func (r *resourceService) ModifyPlan(ctx context.Context, req resource.ModifyPla
...
-
-		properties := []*types.String{
-			&kpi.Description, &kpi.ThresholdTemplateID,
-		}
-
-		for _, p := range properties {
-			if p.IsUnknown() {
-				*p = types.StringNull()
+			if kpi.Urgency.IsUnknown() {
+				kpi.Urgency = existingKpi.Urgency
```
Comment (dmitrykruglov) [inline provider/resource_service.go:398]: this `if` seems unnecessary because you are setting Urgency the same way one the line above(396)

**Review by dmitrykruglov — APPROVED** [review]

File: provider/resource_service.go (hunk 5 above)
Comment (masha1997) [reply provider/resource_service.go:398]: oh, thank you, that should be  `=default_urgency`

Comment (masha1997) [reply provider/resource_service.go:398]: its above, yes, just unnecessary

**Pull request merged**